### PR TITLE
Convert AxHost Interfaces to Cswin32

### PR DESCRIPTION
**Pull request by lonitra**
- Removes our definition of `IOleInPlaceActiveObject`, `IOleInPlaceObject`, `IOleInPlaceObjectWindowless`,
`IOleWindow`,`IOleClientSite`,`IOleContainer`,`IOleObject`,`IPerPropertyBrowsing` and use Cswin32 interfaces
- Add helper functions in ComHelpers.cs to reduce redundant code revolving around querying for interface of an object
- Axhost interfaces revolving around streams/storage will be a follow up.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8020)

### CODE REVIEW

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/System/Ole/CADWORD.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+namespace Windows.Win32.System.Ole;
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/System/Ole/CADWORD.cs:7]: nit: blank line after namespace

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.AxContainer.cs [hunk 2]
```diff
@@ -352,7 +354,7 @@ internal IEnumUnknown EnumControls(Control ctl, OLECONTF dwOleContF, GC_WCH dwWh
                         }
                     }
 
-                    return new EnumUnknown(rval);
+                    return (IEnumUnknown)new EnumUnknown(rval);
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/AxHost.AxContainer.cs:357]: Is this cast necessary?

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.cs [hunk 3]
```diff
@@ -1221,28 +1222,28 @@ private static int Twip2Pixel(int v, bool xDirection)
...
             }
 
             return GetExtent();
         }
 
         private unsafe Size GetExtent()
         {
-            var sz = new Size();
-            GetOleObject().GetExtent(Ole32.DVASPECT.CONTENT, &sz);
+            Size sz = new();
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/AxHost.cs:1245]: nit: `size`

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/Com2IPerPropertyBrowsingHandler.Com2IPerPropertyBrowsingEnum.cs [hunk 4]
```diff
@@ -90,7 +90,7 @@ private unsafe void EnsureArrays()
                         }
 
                         using VARIANT variant = new();
-                        HRESULT hr = ppb.GetPredefinedValue(_target.DISPID, cookie, &variant);
+                        HRESULT hr = ppb.GetPredefinedValue((int)_target.DISPID, cookie, (VARIANT*)&variant);
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/Com2IPerPropertyBrowsingHandler.Com2IPerPropertyBrowsingEnum.cs:93]: Is the cast necessary?

**Review by JeremyKuhne — DISMISSED** [review]
Just some nits, otherwise looks good!

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.AxContainer.cs (hunk 2 above)
Comment (lonitra) [reply src/System.Windows.Forms/src/System/Windows/Forms/AxHost.AxContainer.cs:357]: Oops. Should've gotten rid of our own `IEnumUnknown` definition here.

**Review by RussKie — COMMENTED** [review]
👍

**Review by JeremyKuhne — APPROVED** [review]

**Pull request merged**